File: sdkwork-ai-sdk-swift/Sources/API/EmbeddingsApi.swift
import Foundation

public class EmbeddingsApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create embeddings
    public func createEmbedding(body: EmbeddingRequest) async throws -> EmbeddingResponse? {
        let response = try await client.post(ApiPaths.aiPath("/embeddings"), body: body)
        return response as? EmbeddingResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/EmbeddingsApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class EmbeddingsApi
    {
        private readonly HttpClient _client;

        public EmbeddingsApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Create embeddings
        /// </summary>
        public async Task<EmbeddingResponse?> CreateEmbeddingAsync(EmbeddingRequest body)
        {
            return await _client.PostAsync<EmbeddingResponse>(ApiPaths.AiPath("/v1/embeddings"), body);
        }
    }
}
